R/unreachable_code_linter.R

```diff
@@ -104,7 +112,19 @@ unreachable_code_linter <- function() {
     lints_return_stop <- xml_nodes_to_lints(
       expr_return_stop[!is_nolint_end_comment],
       source_expression = source_expression,
-      lint_message = "Code and comments coming after a top-level return() or stop() should be removed.",
+      lint_message = "Code and comments coming after a return() or stop() should be removed.",
+      type = "warning"
+    )
+
+    expr_next_break <- xml_find_all(xml, xpath_next_break)
+
+    is_nolint_end_comment <- xml2::xml_name(expr_next_break) == "COMMENT" &
```

Comment:
I would add a helper like drop_nolint_end_comment to filter the initial match since it's re-used. that way it's easier to keep the logic in sync & modular.